### PR TITLE
NMS-14447: Update Jira ticketing plugin doc

### PR DESCRIPTION
**Pull request by emarsh1995**
Note: this PR was created because of issues with the original NMS-14447 branch. The previous PR (https://github.com/OpenNMS/opennms/pull/5118) has been closed.

------

Copy-pasted changes from original NMS-14447 branch.

- Updated and reformatted reference/ticketing/ticketer/jira.adoc
- Added requested lines to document

### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here, or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready. Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-14447

### CODE REVIEW

File: docs/modules/reference/pages/ticketing/ticketer/jira.adoc [hunk 1]
```diff
@@ -259,12 +272,12 @@ custom checkbox                customfield_10100    true       array <6>
...
-<3> Defined issue types are `Bug` and `Task`.
-<4> Defined priorities are `Major` and `Minor`.
-<5> Defined projects are `NMS` and `HZN`.
-<6> Defined options are `yes`, `no`, and `sometimes`.
+. Defined components are `core`, `service`, and `web`.
+. Defined versions are `1.0.0` and `1.0.1`.
+. Defined issue types are `Bug` and `Task`.
+. Defined priorities are `Major` and `Minor`.
+. Defined projects are `NMS` and `HZN`.
+. Defined options are `yes`, `no`, and `sometimes`.
```
Comment (mmahacek) [inline docs/modules/reference/pages/ticketing/ticketer/jira.adoc:281]: Not sure this section is needed as the following example re-states these.

Comment (emarsh1995) [reply docs/modules/reference/pages/ticketing/ticketer/jira.adoc:281]: In that case, should we completely remove this example?

File: docs/modules/reference/pages/ticketing/ticketer/jira.adoc [hunk 2]
```diff
@@ -2,119 +2,128 @@
...
 | Name of the transition to use when resolving issues.
+The transition name is typically found on a button when looking at the ticket in Jira.
 
 | jira.reopen
 | Name of the transition to use when reopening issues.
+The transition name is typically found on a button when looking at the ticket in Jira.
 
 | jira.status.open
-| Comma-separated list of Jira status names for which the ticket should be considered 'Open'.
+| A comma-separated list of Jira status names under which a ticket should be considered 'Open.'
```
Comment (Bonrob2) [inline docs/modules/reference/pages/ticketing/ticketer/jira.adoc:65]:
```suggestion
| A comma-separated list of Jira status names that classify a ticket as "Open."
```

File: docs/modules/reference/pages/ticketing/ticketer/jira.adoc [hunk 3]
```diff
@@ -2,119 +2,128 @@
...
 
 | jira.status.closed
-| Comma-separated list of Jira status names for which the ticket should be considered 'Closed'.
+| A comma-separated list of Jira status names under which a ticket should be considered 'Closed.'
```
Comment (Bonrob2) [inline docs/modules/reference/pages/ticketing/ticketer/jira.adoc:68]:
```suggestion
| A comma-separated list of Jira status names that classify a ticket as "Closed."
```

File: docs/modules/reference/pages/ticketing/ticketer/jira.adoc [hunk 4]
```diff
@@ -2,119 +2,128 @@
...
 
 | jira.status.cancelled
-| Comma-separated list of Jira status names for which the ticket should be considered 'Canceled'.
+| A comma-separated list of Jira status names under which a ticket should be considered 'Canceled.'
```
Comment (Bonrob2) [inline docs/modules/reference/pages/ticketing/ticketer/jira.adoc:71]:
```suggestion
| A comma-separated list of Jira status names that classify a ticket as "Canceled."
```

File: docs/modules/reference/pages/ticketing/ticketer/jira.adoc [hunk 5]
```diff
@@ -2,119 +2,128 @@
...
+
+To list all available commands, type `help | grep jira` in the Karaf shell.
+
+Include `--help` after a command to see that command's documentation.
 
 == Custom fields
 
-Besides the common fields creator, create date, description, or subject, you also may also need to set ticket-system proprietary fields.
-The Jira ticketing plugin provides the opportunity to define those in the OpenNMS ticket attributes, which can be overwritten with Drools rules.
+Besides the common fields (creator, create date, description, and subject), you also may also need to set custom fields.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/ticketing/ticketer/jira.adoc:102]:
```suggestion
Besides the common fields (creator, create date, description, and subject), you also may need to set custom fields.
```

File: docs/modules/reference/pages/ticketing/ticketer/jira.adoc [hunk 6]
```diff
@@ -131,17 +140,18 @@ salience 100
...
-You may need to set Jira default values as well; for example, the Component, the Reporter, or the Assignee.
-You can even define the project key or issue type differently than originally in the `jira.properties`.
+You can show all fields by appending the `-s` option to the command.
 
-The OpenNMS ticketer attribute model only lets you set a string value.
-However, the Jira model is slightly different.
-Therefore, each string value must be converted to a Jira field type.
-The following table describes valid values for an OpenNMS attribute.
+You may also need to set default values (for example, the component, reporter, or assignee).
+The project key and issue types can also be redefined in the `jira.properties` file.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/ticketing/ticketer/jira.adoc:149]:
```suggestion
You can also redefine the project key and issue types in the `jira.properties` file.
```

File: docs/modules/reference/pages/ticketing/ticketer/jira.adoc [hunk 7]
```diff
@@ -150,91 +160,94 @@ The following table describes valid values for an OpenNMS attribute.
 | Description
 
 | any
-| Any string
+| Any String.
 
 | date
-| Any date in the format YYYY-MM-DD
+| Any date in the _YYYY-MM-DD_ format.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/ticketing/ticketer/jira.adoc:166]:
```suggestion
| Any date in YYYY-MM-DD format.
```

File: docs/modules/reference/pages/ticketing/ticketer/jira.adoc [hunk 8]
```diff
@@ -150,91 +160,94 @@ The following table describes valid values for an OpenNMS attribute.
...
 
 | datetime
-| Any date-time in ISO 8601 format: YYYY-MM-DDThh:mm:ss.sTZD.
+| Any date-time value in the ISO 8601 format (_YYYY-MM-DDThh:mm:ss.sTZD_).
```
Comment (Bonrob2) [inline docs/modules/reference/pages/ticketing/ticketer/jira.adoc:169]:
```suggestion
| Any date-time value in the ISO 8601 format (YYYY-MM-DDThh:mm:ss.sTZD).
```

File: docs/modules/reference/pages/ticketing/ticketer/jira.adoc (hunk 1 above)
Comment (Bonrob2) [reply docs/modules/reference/pages/ticketing/ticketer/jira.adoc:281]: Should we keep the output example, but not the legend that explains everything?

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
A few minor changes. 

I notice that the class is "Ticketer" plugin rather than "ticketing", and we do refer to "ticketer" when discussing the class. @mmahacek do you know if there is a reason why it has been called "ticketing" plugin in the docs? Should we change it to "ticketer" plugin?

**Comment by mmahacek** [comment]
I don't have a definitive answer on the ticketing terms.  Generally speaking, I think that the "ticketing" system uses one of the "ticketer" classes to work with a specific 3rd party system.

**Comment by Bonrob2** [comment]
> I don't have a definitive answer on the ticketing terms. Generally speaking, I think that the "ticketing" system uses one of the "ticketer" classes to work with a specific 3rd party system.

That makes sense. Let's leave it as "ticketing". It does make sense, just wanted to check we weren't missing something.

**Review by mmahacek — APPROVED** [review]
LGTM

**Review by Bonrob2 — APPROVED** [review]
LGTM2!

**Comment by emarsh1995** [comment]
I'll just remove the legend from that example, then it looks like we'll be good to go!

File: docs/modules/reference/pages/ticketing/ticketer/jira.adoc (hunk 1 above)
Comment (emarsh1995) [reply docs/modules/reference/pages/ticketing/ticketer/jira.adoc:281]:
```suggestion

```

**Review by mmahacek — APPROVED** [review]
lgtm

**Review by Bonrob2 — APPROVED** [review]
LGTM!

**Pull request merged**